### ISSUE TITLE
arxiv paper titles not visible on hypothes.is annotations page

### ISSUE THREAD

**Issue by dbieber** [hypothesis/h#7868]
tl;dr It would be great to be able to modify the name of a document/page, since pdfs from arxiv show up with just their id (e.g. 1607.06450v1.pdf)

Hello! I hope this is an OK place for a feature request. I use hypothesis to take notes on pdfs from arxiv.org, but when I review my notes on hypothes.is it's difficult to see what each pdf/page is. The list of pdfs/pages looks like this:
<img width="906" alt="image" src="https://user-images.githubusercontent.com/892765/221947952-b0f20089-7e55-45ed-80c4-ffd749e3c869.png">
As you can see, it just shows the pdf arxiv id, not the paper title! So, my feature request is: can we add a way to modify the title of a document so this notes overview page becomes more readable?

**Comment by kael** [comment]
You could write a Greasemonkey script which parses the Arxiv PDF links on Hypothesis pages, gathers their [ArxivID](https://info.arxiv.org/help/arxiv_identifier.html), queries the Arxiv ATOM API with one HTTP call (using an [aggregated list of ArxivIDs as a search parameter](https://info.arxiv.org/help/api/user-manual.html#3111-search_query-and-id_list-logic)) and updates Arxiv PDF links titles on Hypothesis pages.

Example of an API query with multiple ArxivIDs: https://export.arxiv.org/api/query?id_list=2007.12026,1706.03741,1603.00022

**Comment by dbieber** [comment]
Thanks for the suggestion. With considerable help from GPT-4, I implemented your suggestion as a Tampermonkey script. It works great! Here's the script: https://gist.github.com/dbieber/c3198a1ceeb86fa823df305a76907afa

**Comment by kael** [comment]
> With considerable help from GPT-4, I implemented your suggestion as a Tampermonkey script.

You made my day. :smile: :+1:
